trackmania.js

```diff
@@ -278,6 +278,24 @@ export class CoreService extends BaseNadeoService {
         }
         return data;
     }
+
+    /**
+     * 
+     * @param {*} accountIdList 
+     * @param {*} mapId 
+     * @param {*} gameMode 
+     * @param {*} seasonId 
+     * @returns {Promise<any[]>}
+     */
+    async getMapRecords(accountIdList, mapId, gameMode, seasonId = undefined) {
+        console.log(gameMode);
+        let endpoint = '/v2/mapRecords/?accountIdList=' +
+        accountIdList.join(',') + '&mapId=' + mapId;
+        if (seasonId !== undefined) endpoint +=  `&seasonId=${seasonId}`;
+        if (gameMode !== undefined && gameMode !== 'Race') endpoint += `&gameMode=${gameMode}`;
+        console.log(endpoint);
+        return await this.fetchEndpoint('getMapRecords', endpoint);
```

Comment:
noice. Not actionable in this PR, but if we start adding optional params to other endpoints, then maybe we can change `fetchEndpoint` to accept a `param` object and let it handle the logic of adding query params + filtering undefined ones out

```suggestion
        const endpoint = '/v2/mapRecords';
        const params = {
            accountIdList: accountIdList.join(','),
            mapId,
            seasonId,
            gameMode: gameMode === 'Race' ? undefined : gameMode,
        };
        log.info(`GetMapRecords params: ${JSON.stringify(params)}`);
        return await this.fetchEndpoint('getMapRecords', endpoint, params);
```